File: batches_java/batch_10_code_reordering/ref.java
public class Stats {
    public static double[] computeStats(int[] data) {
        int n = data.length;
        int mx = data[0], mn = data[0], total = 0;
        for (int v : data) {
            if (v > mx) mx = v;
            if (v < mn) mn = v;
            total += v;
        }
        double avg = n > 0 ? (double) total / n : 0;
        return new double[]{avg, mx, mn};
    }

    public static void main(String[] args) {
        int[] d = {10, 20, 30, 40, 50};
        double[] result = computeStats(d);
        System.out.printf("avg=%.1f max=%.0f min=%.0f%n", result[0], result[1], result[2]);
    }
}

File: batches_c/batch_10_code_reordering/ref.c
#include <stdio.h>

void compute_stats(int *data, int len, double *avg, int *mx, int *mn) {
    *mx = data[0]; *mn = data[0];
    int total = 0;
    for (int i = 0; i < len; i++) {
        if (data[i] > *mx) *mx = data[i];
        if (data[i] < *mn) *mn = data[i];
        total += data[i];
    }
    *avg = len > 0 ? (double)total / len : 0;
}

int main() {
    int d[] = {10, 20, 30, 40, 50};
    double avg; int mx, mn;
    compute_stats(d, 5, &avg, &mx, &mn);
    printf("avg=%.1f max=%d min=%d\n", avg, mx, mn);
    return 0;
}
